backend/src/features/profiles/dto/create-profile.dto.ts

```diff
@@ -1,18 +1,88 @@
-import { ApiExtraModels, ApiProperty, PartialType } from '@nestjs/swagger';
+import { ApiExtraModels, ApiProperty } from '@nestjs/swagger';
 import { Profile } from 'src/schemas/profile.schema';
-import { IsString } from 'class-validator';
+import {
+  IsNumber,
+  IsOptional,
+  IsString,
+  ValidateNested,
+} from 'class-validator';
+import { Type } from 'class-transformer';
 
+class MetaDataDTO {
+  @IsNumber()
+  @ApiProperty({ example: 12 })
+  totalCoin: number;
+
+  @IsNumber()
+  @ApiProperty({ example: 100 })
+  missionSuccess: number;
+}
+
+class CharacterDTO {
+  @IsString()
+  @ApiProperty({ example: 'id' })
+  id: string;
+
+  @IsString()
+  @ApiProperty({ example: 'name' })
+  name: string;
+
+  @IsString()
+  @ApiProperty({ example: 'photo_url' })
+  photoUrl: string;
+
+  @IsString()
+  @ApiProperty({ example: 'detail' })
+  detail: string;
+}
 @ApiExtraModels(Profile)
-export class CreateProfileDto extends PartialType(Profile) {
+export class CreateProfileDto {
   @IsString()
-  @ApiProperty({ example: 'mobi' })
+  @ApiProperty({ example: 'display_name' })
+  displayName: string;
+
+  @IsString()
+  @IsOptional()
+  @ApiProperty({ example: 'first_name' })
   firstName: string;
 
   @IsString()
-  @ApiProperty({ example: 'le' })
+  @IsOptional()
+  @ApiProperty({ example: 'last_name' })
   lastName: string;
 
   @IsString()
-  @ApiProperty({ example: 'mobile@gmail.com' })
+  @IsOptional()
+  @ApiProperty({ example: 'example@mail.com' })
   email: string;
+
+  @IsString()
+  @IsOptional()
+  @ApiProperty({ example: 'female' })
+  gender: string;
```

Comment:
gender should be enum "MALE" "FEMALE" "OTHER"

Reply:
fixed